docs/recipes/endpoint-testing-with-mongoose.md

```diff
@@ -0,0 +1,281 @@
+```
+
+You should have Mongoose installed already. If not, run the following code to install it:
+
+(Note: You need at least mongoose v4.11.3)
+
+```console
+$ npm install mongoose
+```
+
+## Setting up AVA
+
+Since MongoDB Memory server requires Babel polyfill to work, the easiest way to set up AVA is through the `ava` key in your `package.json` file.
+
+```json
+"ava": {
+    "require": [
+      "babel-register",
+      "babel-polyfill"
+    ]
+  },
+```
+
+If you want AVA to run your tests whenever changes are made, add the `--watch` option.
+
+```json
+"scripts": {
+  "watch": "ava --watch"
+},
+```
+
+## Your test file
+
+First, include the libraries you need.
+
+```js
+// Libraries required for testing
+import test from 'ava'
+import request from 'supertest'
+import MongodbMemoryServer from 'mongodb-memory-server'
+import mongoose from 'mongoose'
+
+// Your server and models
+import app from '../server'
+import User from '../models/User'
+```
+
+### Your server file
+
+In your `server` file, you cannot use `app.listen` to start your app. Export it to another file and call `app.listen` there.
+
+This `server` file should resemble the following:
+
+```js
+const express = require('express')
+const bodyParser = require('body-parser')
+const routes = require('./routes')
+const app = express()
+
+// ======================================
+// # Middlewares
+// ======================================
+app.use(bodyParser.json())
+
+// ======================================
+// # Routes
+// ======================================
+app.get('/litmus', async (req, res) => {
+  const { email } = req.body
+  res.json(await User.findOne({email}))
+})
+
+app.post('/litmus', async (req, res) => {
+  const { email, name } = req.body
+  const user = new User({email, name})
+  res.json(await user.save())
+})
+
+module.exports = app
+```
+
+### Back to your test file
+
+**First, start your MongoDB instance and connect to Mongoose:**
+
+```js
+// Start MongoDB Instance
+const mongod = new MongodbMemoryServer()
+
+// Create connection to mongoose before all tests
+test.before(async t => mongoose.connect(await mongod.getConnectionString(), { useMongoClient: true }))
+```
+
+When you run your first test, MongoDB downloads the latest MongoDB Binaries. It may take a minute. (The download is ~70mb).
+
+Note: Since we're using async/await, you need Node v7.6 and above.
+
+**Add fixtures for each test**
+
+You'll want to populate your database with dummy data. Here's an example:
+
+```js
+test.beforeEach(async t => {
+  const user = new User({
+  	email: 'one@example.com',
+  	name: 'One'
+  })
+  await user.save()
+})
+```
+
+**Clear your dummy data after each test**:
+
+```js
+// Cleans up database after every test
+test.afterEach.always(async t => await User.remove())
+```
+
+**Write your tests**
+
+Use Supertest to fire a request for your endpoint. Then, do the rest with AVA normally.
+
+Note: Make sure your tests run serially with `test.serial`.
```

Comment:
It should be summarized in the recipe. Otherwise people may either think that they should always use `test.serial()`, or they'll go "that can't be right" and remove it.